src/linux_mcp_server/tools/storage.py

```diff
@@ -83,287 +98,155 @@ async def list_block_devices(
+
+    match order_by:
+        case OrderBy.SIZE:
+            # Use du command to get directory sizes efficiently
+            returncode, stdout, _ = await execute_command(
+                ["du", "-b", "--max-depth=1", path],
+                host=host,
+                username=username,
+            )
+
+            # Parse output - du may return non-zero on permission errors but still give valid data
+            lines = stdout.strip().split("\n")
+            directories_by_size: list[tuple[int, str]] = []
+
+            for line in lines:
+                if not line:
+                    continue
                 parts = line.split("\t", 1)
                 if len(parts) == 2:
                     try:
-                        timestamp = float(parts[0])
-                        dir_name = parts[1]
-                        directories.append((timestamp, dir_name))
-                    except ValueError:
+                        size = int(parts[0])
+                        dir_path_str = parts[1]
+                        # Skip the parent directory itself
+                        dir_name = Path(dir_path_str).name
+                        if dir_path_str != path:
+                            directories_by_size.append((size, dir_name))
+                    except (ValueError, IndexError):
                         continue
 
-        if not directories:
-            return f"No subdirectories found in: {path}"
+            if not directories_by_size:
+                # Only error if we got no output AND a bad return code
+                if returncode != 0:
+                    raise ToolError("du command failed and returned no directory data")
+                raise ToolError(f"No subdirectories found in: {path}")
+
+            # Sort by size
+            reverse = sort == SortBy.DESCENDING
+            directories_by_size.sort(key=lambda x: x[0], reverse=reverse)
+
+            return directories_by_size
+        case OrderBy.NAME:
+            # Use find to list only immediate subdirectories
+            returncode, stdout, _ = await execute_command(
+                ["find", path, "-mindepth", "1", "-maxdepth", "1", "-type", "d", "-printf", "%f\\n"],
+                host=host,
+                username=username,
+            )
 
-        # Sort by timestamp
-        directories.sort(key=lambda x: x[0], reverse=newest_first)
+            if returncode != 0:
+                raise ToolError(f"Error running find command: command failed with return code {returncode}")
 
-        # Format output
-        result = []
-        sort_order = "Newest First" if newest_first else "Oldest First"
-        result.append(f"=== Directories ({sort_order}) ===")
-        result.append(f"Path: {path_str}")
-        result.append(f"\nTotal subdirectories found: {len(directories)}\n")
+            # Parse and sort output
+            directories_by_name: list[tuple[None, str]] = [(None, line) for line in stdout.strip().split("\n") if line]
```

Comment:
What's the purpose of creating a list of (None, 'something') pairs? Why not just create a list of directory names?